misc/extensions.talon

```diff
@@ -1 +1,2 @@
+-
```

Comment:
Adding the `-` when there is no context header was a bit of a sticking point in our grooming session

Reply:
I'm guessing that removing the `-` when there is no context header will result in a smaller diff, so maybe worth doing that and then filing a follow-up issue to discuss always having the `-`, in the interest of getting this PR merged sooner

Reply:
Cursorless support will be better if it’s mandatory, since with it optional the grammar is ambiguous.

Reply:
Yes, we do want to remove auto-adding the `-`.

Reply:
(Tangent, but: I'm reasonably sure it's not actually ambiguous, just hard to express in an LR formalism. Can you give an example of something that has multiple parses = is ambiguous?)

Reply:
It’s not ambiguous, but it’s not parseable but a non-backtracking parser.

Reply:
Do you all want to keep the `-` if it's already there, even if it's not needed?